api/src/main/java/ca/bc/gov/educ/studentdatacollection/api/rules/impl/IndigenousRule.java

```diff
@@ -29,18 +29,22 @@ public boolean shouldExecute(SdcStudentSagaData sdcStudentSagaData) {
     @Override
     public List<SdcSchoolCollectionStudentValidationIssue> executeValidation(SdcStudentSagaData sdcStudentSagaData) {
         final List<SdcSchoolCollectionStudentValidationIssue> errors = new ArrayList<>();
+        var student = sdcStudentSagaData.getSdcSchoolCollectionStudent();
         List<BandCode> activeBandCodes = validationRulesService.getActiveBandCodes();
 
-        if(!sdcStudentSagaData.getSdcSchoolCollectionStudent().getNativeAncestryInd().equals("Y") && !sdcStudentSagaData.getSdcSchoolCollectionStudent().getNativeAncestryInd().equals("N")) {
+        if(!student.getNativeAncestryInd().equals("Y") && !student.getNativeAncestryInd().equals("N")) {
             errors.add(createValidationIssue(SdcSchoolCollectionStudentValidationIssueSeverityCode.ERROR, SdcSchoolCollectionStudentValidationFieldCode.NATIVE_ANCESTRY_IND, SdcSchoolCollectionStudentValidationIssueTypeCode.NATIVE_IND_INVALID));
         }
 
-        if(StringUtils.isNotEmpty(sdcStudentSagaData.getSdcSchoolCollectionStudent().getBandCode()) && (StringUtils.isEmpty(sdcStudentSagaData.getSdcSchoolCollectionStudent().getSchoolFundingCode()) || !sdcStudentSagaData.getSdcSchoolCollectionStudent().getSchoolFundingCode().equals(Constants.IND_FUNDING_CODE))) {
+        if((StringUtils.isEmpty(student.getBandCode()) && (StringUtils.isNotEmpty(student.getSchoolFundingCode())
+                && student.getSchoolFundingCode().equals(Constants.IND_FUNDING_CODE))) ||
+                (StringUtils.isNotEmpty(student.getBandCode()) && (StringUtils.isEmpty(student.getSchoolFundingCode())
+                || !student.getSchoolFundingCode().equals(Constants.IND_FUNDING_CODE)))) {
             errors.add(createValidationIssue(SdcSchoolCollectionStudentValidationIssueSeverityCode.ERROR, SdcSchoolCollectionStudentValidationFieldCode.BAND_CODE, SdcSchoolCollectionStudentValidationIssueTypeCode.BAND_CODE_BLANK));
             errors.add(createValidationIssue(SdcSchoolCollectionStudentValidationIssueSeverityCode.ERROR, SdcSchoolCollectionStudentValidationFieldCode.SCHOOL_FUNDING_CODE, SdcSchoolCollectionStudentValidationIssueTypeCode.BAND_CODE_BLANK));
```

Comment:
It looks like we're adding a "BAND_CODE_BLANK" to a "SCHOOL_FUNDING_CODE" field.  Is this goofy?